File: HW/HW1/IsPalindrome.py
def is_palindrome(s):
    s = s.lower()
    stack = []
    regular = ""
    reversed = ""
    for char in s:
        if char.isalnum():
            stack.append(char)
            regular += char
    while len(stack)>0:
        reversed += stack.pop()
    if (regular==reversed):
        return True
    return False
print(is_palindrome("raaah"))
print(is_palindrome("Bub"))
print(is_palindrome("a@aa"))
print(is_palindrome(""))

File: HW/HW1/IsPalindrome.java
package HW1;

import java.util.Stack;

public class IsPalindrome {
    public static boolean is_palindrome(String s){
        String alpha = "";
        String reversed = "";
        Stack<Character> stack = new Stack<Character>();
        for (int i = 0; i < s.length(); i++){
            if (Character.isLetter(s.charAt(i)) || (Character.isDigit(s.charAt(i)))){
                stack.push(s.charAt(i));
                alpha += s.charAt(i);
            }
        }
        while (!stack.isEmpty()){
            Character popped = stack.pop();
            reversed += popped;
        }
        if (alpha.equalsIgnoreCase(reversed))
            return true;
        return false;
    }
    public static void main(String[] args){
        System.out.println(is_palindrome("AAAAH123!"));
        System.out.println(is_palindrome("JIJ"));
        System.out.println(is_palindrome("121!"));
    }
}
